app/models/invoice.rb

```diff
@@ -501,10 +502,8 @@ def not_paid?
   CONDITIONS_TO_CHARGE = %i[not_paid provider_present provider_payment_gateway_configured positive buyer_account_paying_monthly].freeze
 
   def reason_cannot_charge
-    CONDITIONS_TO_CHARGE.each do |condition_method|
-      return I18n.t(condition_method, scope: %i[invoices reasons_cannot_charge]) unless method("#{condition_method}?").call
-    end
-    nil
+    reason = CONDITIONS_TO_CHARGE.find { |condition| !method("#{condition}?").call }
+    I18n.t(reason, scope: %i[invoices reasons_cannot_charge]) if reason
```

Comment:
What _if not_ reason? Does it return null.. ?

Reply:
it does

Reply:
maybe it would be clearer with:
```ruby
return unless (reason = CONDITIONS_TO_CHARGE.find { |condition| !method("#{condition}?").call })
I18n.t(reason, scope: %i[invoices reasons_cannot_charge])
```
But either way is fine 👍

Reply:
I didn't even have to touch this, TBH. Just did to make it shorter. It's Ruby language anyway.